yb-voyager/cmd/importDataFileCommand.go

```diff
@@ -322,6 +326,68 @@ func checkAndParseEscapeAndQuoteChar() {
 
 }
 
+func packAndSendImportDataFilePayload(status string) {
+	if !callhome.SendDiagnostics {
+		return
+	}
+	payload := createCallhomePayload()
+	payload.MigrationType = BULK_DATA_LOAD
+	bytes, err := json.Marshal(targetDBDetails)
+	if err != nil {
+		log.Errorf("callhome: error in parsing sourcedb details: %v", err)
+	}
+	payload.TargetDBDetails = string(bytes)
+	payload.MigrationPhase = IMPORT_DATA_FILE_PHASE
+	dataFileParameters := callhome.DataFileParameters{
+		FileFormat: fileFormat,
+		HasHeader:  bool(hasHeader),
+		Delimiter:  delimiter,
+		EscapeChar: escapeChar,
+		QuoteChar:  quoteChar,
+		NullString: nullString,
+	}
+	parametersBytes, err := json.Marshal(dataFileParameters)
+	if err != nil {
+		log.Errorf("callhome: error in parsing dataFileParameters: %v", err)
+	}
```

Comment:
@priyanshi-yb since `DataFileParameters` is a struct, how about defining a method in it.
`func (dfp *DataFileParameters) MarshalledJsonString() string`
or some better method name....

which can be directly used:
```
importDataFilePayload := callhome.ImportDataFilePhasePayload{
		ParallelJobs:       int64(tconf.Parallelism),
		StartClean:         bool(startClean),
		DataFileParameters: dataFileParameters.MarshalledJsonString(),
	}
```

You can similarly do this for other structs also, this will abstract out the code at the cmd.go level.

Reply:
Yeah, thats nice. Thanks. I have implemented a single generic function for all types to marshall.